src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AvoidNoArgumentSuperConstructorCallCheck.java

```diff
@@ -0,0 +1,93 @@
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks if call to superclass constructor without arguments is present.
+ * Such invocation is redundant because constructor body implicitly
+ * begins with a superclass constructor invocation {@code super();}
+ * See <a href="https://docs.oracle.com/javase/specs/jls/se13/html/jls-8.html#jls-8.8.7">
+ * specification</a> for detailed information.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AvoidNoArgumentSuperConstructorCall&quot;/&gt;
+ * </pre>
+ * <p>
+ * Example of violations
+ * </p>
+ * <pre>
+ * class MyClass extends SomeOtherClass {
+ *     MyClass() {
+ *         super(); // violation
+ *     }
+ *
+ *     MyClass(int arg) {
+ *         super(arg); // OK, call with argument should be explicit
+ *     }
+ *
+ *     MyClass(long arg) {
+ *         // OK, call is implicit
+ *     }
+ * }
+ * </pre>
+ * @since 8.28
+ */
+@StatelessCheck
+public final class AvoidNoArgumentSuperConstructorCallCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_CTOR = "super.constructor.call";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.SUPER_CTOR_CALL,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.findFirstToken(TokenTypes.ELIST).getChildCount() == 0) {
```

Comment:
It can't be so easy. As per [JLS 8.8.7.1](https://docs.oracle.com/javase/specs/jls/se13/html/jls-8.html#jls-ExplicitConstructorInvocation) sometimes `super()` isn't redundant. For example:
```
class Outer {
    class InnerNonStatic {
    }
}

class Derived extends Outer.InnerNonStatic {
    public Derived(Outer s) {
        s.super(); // Not a redundant call
    }
}
```

Reply:
Omg, is there at lease one case where such syntax is used?!

Ok, it looks like it should be checked that there is neither type arguments nor annotations before invocation, and also it is not a part of expression.

Reply:
I think not part of expression is enough, ( but what about `this.super()` ??? it should be compile error, please recheck).

Reply:
> what about this.super() ??? it should be compile error

It is.

Reply:
@pbludov corrected